test/framework/clusterctl/e2e_config.go

```diff
@@ -519,6 +520,11 @@ func (c *E2EConfig) RuntimeExtensionProviders() []string {
 	return c.getProviders(clusterctlv1.RuntimeExtensionProviderType)
 }
 
+// AddonProviders returns the add-on provider selected for running this E2E test.
+func (c *E2EConfig) AddonProviders() []string {
```

Comment:
I cannot mark the correct file in this PR but you also need to modify [these lines](https://github.com/ykakarap/cluster-api/blob/53ed0e0e590fd1ea01fbaa3ab67f7220915eef62/test/framework/clusterctl/repository.go#L124-L126) in `test/framework/clusterctl/repository.go` to include the addon provider.